pages/indicator.js

```diff
@@ -1,27 +1,132 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+// Modules
+import { getSpecificIndicators } from 'modules/indicators';
+import { setSingleMapParams, setSingleMapParamsUrl } from 'modules/single-map';
+
 // Redux
 import withRedux from 'next-redux-wrapper';
 import { store } from 'store';
 
 // Components
 import Page from 'components/layout/page';
 import Layout from 'components/layout/layout';
+import Map from 'components/map/map';
+
+// Constants
+import { GENERIC_ZINDEX, MAP_OPTIONS } from 'constants/map';
+
+
+// let L;
+// if (typeof window !== 'undefined') {
+//   /* eslint-disable global-require */
+//   L = require('leaflet/dist/leaflet');
+//   /* eslint-enable global-require */
+// }
 
 class IndicatorPage extends Page {
+  constructor(props) {
+    super(props);
+
+    this.update = true;
+  }
+
+  componentDidMount() {
+    const { url } = this.props;
+
+    this.props.getSpecificIndicators(url.query.indicators);
+
+    if (url.query.zoom || url.query.lat || url.query.lng) {
+      this.setParams();
+    }
+  }
+
+  setParams() {
```

Comment:
It would be better if you move this function to the module, so we can see all the functions in the same place